### DISCUSSION TITLE
patterns/event-queue/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#52] [Announcements]
# patterns/event-queue/

Intent The intent of the event queue design pattern, also known as message queues, is to decouple the relationship between the sender and receiver of events within a system. By decoupling the two parties, they do not interact with the event queue simultaneously. Essentially, the event queue handles and processes requests in an asynchronous manner, therefore, this system can be described as a first in, first out design pattern model. Event Queue is a suitable pattern if there is a resource with limited accessibility (i.e. Audio or Database), however, you need to provide access to all the requests which seeks this resource. Upon accessing an event from the queue, the program also removes it from the queue.

https://java-design-patterns.com/patterns/event-queue/

**Comment by huymaster** [comment]
**how to make the queue pause when not have any event ?**

**Reply by iluwatar** [reply]
If we are talking specifically about `Audio.java` included in the example, I guess it would be possible to add autostop functionality. Something like calling `Audio#stopService()` automatically when the queue becomes empty.
